logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -51,12 +51,12 @@ def fetch_or_store(namespaces, key, default_value = nil)
       # BUT. If the value is not present in the `@fast_lookup` the value will be inserted and we assume that we don't
       # have it in the `@metric_store` for structured search so we add it there too.
 
-      value = @fast_lookup.get(namespaces.dup << key)
+      value = @fast_lookup.get((namespaces.dup << key).hash)
```

Comment:
@jakelandis 

>  but I believe this is safe enough

Phew :D based on what? :D Tbh I have no way to estimate this either, but this seems like a potential source for very strange bugs.
Frankly, I don't have anything better on this, but imo it's probably worth going with a strong hash function and a trie type hash here since we're not under a lot of time pressure (+ this shouldn't be that hard to implement imo). 
But ... your call, my argument sucks without math to back it up and I admit it :)